### PR TITLE
Add SSR support

### PR DESCRIPTION
**Pull request by therealparmesh**
This attempts to solve the need identified in https://github.com/angelle-sw/use-axios-client/issues/28 by adding and implementing the `ssrData` prop.

To do:
- [x] Update tests
- [x] Update docs

### CODE REVIEW

File: src/useBaseAxios.ts [hunk 1]
```diff
@@ -8,26 +8,29 @@ interface RequestFunctions {
   refetch: () => Promise<void>;
 }
 
-export interface Config extends AxiosRequestConfig {
+export interface Config<Data> extends AxiosRequestConfig {
   axiosInstance?: AxiosInstance;
+  ssrData?: Data;
```
Comment (zakangelle) [inline src/useBaseAxios.ts]: dig it

File: src/useAxiosReducer.ts [hunk 2]
```diff
@@ -11,11 +11,11 @@ type Action<Data> =
   | { type: 'REQUEST_SUCCESS'; payload: Data }
   | { type: 'REQUEST_FAILED'; payload: Error };
 
-export const initialState = {
-  data: undefined,
+export const createInitialState = <Data>({ data }: { data?: Data } = {}) => ({
+  data: typeof data === 'undefined' ? undefined : data,
```
Comment (zakangelle) [inline src/useAxiosReducer.ts]: Could this just be reduced to `data: data`? (+ object shorthand)

File: src/useAxios.ts [hunk 3]
```diff
@@ -20,8 +20,14 @@ function useAxios<Data>(param1: string | Config, param2: Config = {}) {
 
   const [getData, dataStates] = invokeUseBaseAxios();
 
+  const justMounted = useRef(true);
+
   useEffect(() => {
-    getData();
+    if (typeof config.ssrData === 'undefined' || !justMounted.current) {
```
Comment (zakangelle) [inline src/useAxios.ts]: Thoughts on `config.ssrData === undefined` vs. `typeof config.ssrData === 'undefined'`?

Comment (zakangelle) [reply src/useAxios.ts]: Could you elaborate on the need for `justMounted` ref?

File: src/useAxiosReducer.ts (hunk 2 above)
Comment (therealparmesh) [reply src/useAxiosReducer.ts]: I changed the signature with my latest push.

File: src/useAxios.ts (hunk 3 above)
Comment (therealparmesh) [reply src/useAxios.ts]: I followed this pattern: https://github.com/therealparmesh/use-axios-client/blob/v2.0.0/webpack.config.js#L23

Comment (therealparmesh) [reply src/useAxios.ts]: We need the `justMounted` ref to skip the initial call to useEffect (when the component mounts) if SSR data is passed in. [I copied this logic from the `useEffectAfterMount` hook that Kent C. Dodds wrote here.](https://kentcdodds.com/blog/compound-components-with-react-hooks)

File: docs/src/pages/ServerSideRendering.mdx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+If you're using SSR, you can pass in data fetched on the server to initialize the client by using the `ssrData` option.
+
+## Example
+
+```js
+const { data, error, loading } = useAxios({
+  ssrData: [
+    {
+      id: '55b797ca21414b0cbad9847ebe6f3d64',
+      name: 'Avril Lavigne',
```
Comment (zakangelle) [inline docs/src/pages/ServerSideRendering.mdx]: LMAO

**Pull request merged**